gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -33,7 +33,15 @@ roleselection.cv = {
 
 	---
 	-- @realm server
-	ttt_max_baseroles_pct = CreateConVar("ttt_max_baseroles_pct", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "Maximum amount of different baseroles based on player amount. ttt_max_baseroles needs to be 0")
+	ttt_max_baseroles_pct = CreateConVar("ttt_max_baseroles_pct", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "Maximum amount of different baseroles based on player amount. ttt_max_baseroles needs to be 0"),
+
+	---
+	-- @realm server
+	ttt2_roles_shuffle_players = CreateConVar("ttt2_roles_shuffle_players", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "If not enabled, players will get priority for role selection based on the order they connected."),
+
+	---
+	-- @realm server
+	ttt2_roles_trickle_down_odds = CreateConVar("ttt2_roles_trickle_down_odds", "3", {FCVAR_NOTIFY, FCVAR_ARCHIVE}, "The 1/N odds of a player receiving a role and not being passed up."),
```

Comment:
I'm also not 100% sure we need this. If you add this, you have to explain in detail in the UI how changing this convar affects the selection. Because I think it is hard to understand.

Reply:
Personally, I don't even like having the math.random call in there -- I think the shuffle is doing all the work the RNG really needs to do to consider things.

Should we just only focus on conditions that prevent the role from being selected and pass true by default?

Reply:
That's hard to make a call on as I never worked on the roleselection. Maybe @saibotk has more insight into this. Or @ZenBre4ker?

Reply:
The system is broken in a way anyway. As @saibotk mentioned in another PR, I wouldnt give so many options. I would either always allow or forbid avoiding roles alltogether. I dont like that system anyway, so I would suggest to remove role avoiding alltogether.

About the random call, I believe they tried to make it more random as in the past often the same players in a row got the same role over and over again.

And the trickle system I need to take a look again.

Reply:
I feel the same way, all this role avoidance stuff is rarely used by players and should rather be something a server owner decides upon. This just adds unnecessary complexity and confusion.

Reply:
Yeah checked again, I would just set the tricklerate as before to 3 and check if it is 2, so 1/3 chance, that role gets assigned without the other conditions.